### PR TITLE
Build docs on release

### PR DESCRIPTION
**Pull request by drsm79**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

Build & publish documentation when we make a release.

## Why

Documentation is good, and keeping it in sync with releases is better.

## How

Github action running sphinx.

## Test

Working locally, need to test (& probably fix) the action.

## Context

_Provide a bulleted list of GitHub issues, or any other references (mailing list discussion, etc...) that reviewers can reference
for additional information regarding scope of the pull request._

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Documentation
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build_documentation:
+    name: Run code validation steps
```
Comment (alfinkel) [inline .github/workflows/documentation.yml:9]: I think this should be something like `name: Generate documentation`

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Documentation
```
Comment (alfinkel) [inline .github/workflows/documentation.yml:1]: This'll be the badge name as well.  Do we want to shorten it to something like `docs` (lowercase) to match the other badge styles?

**Comment by alfinkel** [comment]
Things look strange to me.  I see my commit in your branch which I think needs to be fixed but I don't see the change from that commit in the `Files changed` tab.

File: doc-source/notifiers.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@
 Notifiers
 =========
 
-The last phase in a typical compliance-tool check run is the notification
+The last phase in a typical framework check run is the notification
 system.  Multiple notifiers can be targeted as part of this phase  by using
 the ``--notify`` option on the ``compliance --check`` command.  Valid
 notifier options are ``stdout``, ``slack``, ``pagerduty``, ``ghe_issues`` and,
```
Comment (alfinkel) [inline doc-source/notifiers.rst:11]: Change `ghe_issues` to `gh_issues` and also add `findings`.  I'm wondering if removing the options altogether would be a good idea?  This way when new notifiers are added we don't need to remember to update the docs.

File: .github/workflows/documentation.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      run: git worktree add $TMPDIR/af-gh-pages gh-pages
+    - name: Build documentation
+      run: |
+        make docs DOC_TARGET=$TMPDIR/af-gh-pages
+    - name: Commit docs
+      run: |
+        cd $TMPDIR/af-gh-pages
+        git add *
+        git commit -a -m 'Documentation update for release' --no-verify
+        git push origin gh-pages
```
Comment (alfinkel) [inline .github/workflows/documentation.yml:33]: To make GH happy, add a blank line after this.

File: Makefile [hunk 5]
```diff
@@ -38,4 +40,4 @@ test::
 docs:
 	# Build the API docs from the source code - overwrites those files, which are ignored by git
 	sphinx-apidoc -o doc-source compliance
-	sphinx-build doc-source doc
\ No newline at end of file
+	sphinx-build doc-source $(DOC_TARGET)
```
Comment (alfinkel) [inline Makefile:43]: Add a blank line after this

File: doc-source/design-principles.rst [hunk 6]
```diff
@@ -62,13 +62,13 @@ for:
 
     {
       "locker": {
-        "repo_url": "https://github.ibm.com/my-org/my-evidence",
+        "repo_url": "https://github.com/my-org/my-evidence",
```
Comment (alfinkel) [inline doc-source/design-principles.rst:65]: `https://github.com/my-org/my-repo` for consistency.

File: doc-source/design-principles.rst [hunk 7]
```diff
@@ -93,7 +93,7 @@ for:
 
    {
      "locker": {
-       "repo_url": "https://github.ibm.com/my-org/my-evidence",
+       "repo_url": "https://github.com/my-org/my-evidence",
```
Comment (alfinkel) [inline doc-source/design-principles.rst:96]: `https://github.com/my-org/my-repo` for consistency.

File: doc-source/running-on-travis.rst [hunk 8]
```diff
@@ -207,7 +203,7 @@ have stored the official configuration into ``official.json`` file:
 
    {
      "locker": {
-       "repo_url": "https://github.ibm.com/YOUR-ORG/evidence-collector"
+       "repo_url": "https://github.com/YOUR-ORG/evidence-collector"
```
Comment (alfinkel) [inline doc-source/running-on-travis.rst:206]: `https://github.com/my-org/my-repo` for consistency.

File: doc-source/design-principles.rst (hunk 6 above)
Comment (drsm79) [reply doc-source/design-principles.rst:65]: I think https://github.com/my-org/my-evidence-repo would be clearer, since there's the config repo, too

**Comment by alfinkel** [comment]
What do you think about adding a docs workflow badge?

File: doc-source/notifiers.rst [hunk 9]
```diff
@@ -159,8 +160,8 @@ correctly. Provide your GitHub Enterprise id and personal access token in your
 credentials file as shown below::
 
   [github_enterprise]
```
Comment (alfinkel) [inline doc-source/notifiers.rst:162]: `[github]`

File: doc-source/notifiers.rst [hunk 10]
```diff
@@ -159,8 +160,8 @@ correctly. Provide your GitHub Enterprise id and personal access token in your
...
-  username=my-ghe-id
-  token=my-ghe-personal-access-token
+  username=my-gh-id
+  token=my-gh-personal-access-token
 
 GHE Summary Issue by Accreditation
```
Comment (alfinkel) [inline doc-source/notifiers.rst:166]: GH Summary Issue by Accreditation

**Pull request merged**